crates/sui-core/src/rpc_index.rs

```diff
@@ -48,40 +57,62 @@ struct MetadataInfo {
     version: u64,
 }
 
-#[derive(Clone, Copy, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord, Debug)]
+/// Checkpoint watermark type
+#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
+pub enum Watermark {
+    Indexed,
+    Pruned,
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord, Debug)]
 pub struct OwnerIndexKey {
     pub owner: SuiAddress,
+
+    pub object_type: StructTag,
+
+    // If this object is coin-like (eg 0x2::coin::Coin) then this will be the balance of the coin
+    // inverted `!coin.balance` in order to force sorting of coins to be from greatest to least
+    pub inverted_balance: Option<u64>,
```

Comment:
don't have a strong preference here, but curious why is this necessary. Can we just iterate backwards for the coin type?

Reply:
That would require more special casing/logic for handling coin type so i figured it would be easier so that we always just iterate in-order for all queries

Reply:
would iterating backwards for all object types work? Not encouraging such a change, just curious

Reply:
maybe? either is an invariant that needs to be upheld